File: 198-HouseRobber/198-HouseRobber.py
# Last updated: 6/11/2026, 11:35:16 AM
class Solution:
    def rob(self, nums: List[int]) -> int:
        # max_money = [0]
        
        # def count_max(ind, money, dp):
        #     if(ind >= len(nums)):
        #         max_money[0] = max(max_money[0],money)                 
        #         return

        #     count_max(ind+1, money)

        #     count_max(ind+2, money+nums[ind])

        # count_max(0,0)
        # return max_money[0]

        rob1 = 0
        rob2 = 0
        for ind in range(len(nums)):
            temp = max((nums[ind]+rob1),rob2)
            rob1 = rob2
            rob2 = temp
        return temp

File: 198-HouseRobber/198-HouseRobber.cpp
// Last updated: 6/11/2026, 11:35:10 AM
class Solution {
public:
    int dp[101][2];
    int rob(vector<int>& nums) {
        int n = nums.size();
        for(int i=n; i>=0; i--){
            if(i == n){
                dp[i][0] = dp[i][1] = 0;
                continue;
            }
            dp[i][0] = max(dp[i+1][0], dp[i+1][1]);
            dp[i][1] = nums[i]+dp[i+1][0];
        }

        return max(dp[0][0], dp[0][1]);
    }
};